### DISCUSSION TITLE
Map new key to accept first match?

### DISCUSSION THREAD

**Discussion by joalof** [folke/flash.nvim#170] [Q&A] [answered]
When I enter the jumping mode with flash.jump() I can hit <CR> to accept the first highlighted match. I'd like to be able to bind another key for this purpose (specifically <c-j>, which right now cancels the jump), I've rummaged thru the docs but still can't figure out how to add the mapping. Grateful for any help!

**Comment by IndianBoy42** [comment, marked as answer]
It isn't really documented but it seems you can pass an actions table through the config that defines custom actions on certain keys https://github.com/folke/flash.nvim/blob/636bcf84af487d2b8c4a5351d1d123ae3cac2ffe/lua/flash/state.lua#L355

There's just a type definition https://github.com/folke/flash.nvim/blob/636bcf84af487d2b8c4a5351d1d123ae3cac2ffe/lua/flash/state.lua#L20

You can see how treesitter uses it to define incremental selection with , ; 
https://github.com/folke/flash.nvim/blob/636bcf84af487d2b8c4a5351d1d123ae3cac2ffe/lua/flash/plugins/treesitter.lua#L120

**Reply by folke** [reply]
Yes, you can indeed use the actions for this for now. Will document this at some point and probably change it a bit as well.

`opts.actions` should be a table, mapping keycodes to a function.

To map a key to its keycode, you can use the `flash.util` module.

```lua
      opts.actions = {
        [Util.t("<cr>")] = function()
          state:jump() -- jump to first target
          return false -- break the loop
        end
     }
```

**Reply by joalof** [reply]
Thanks for the help:) I noticed that this only works when an actions table is explicitly passed to state:loop, and not when passed e.g., to Command.jump itself. I think the issue might be https://github.com/folke/flash.nvim/blob/7ddeb57f60e2f19ee836aef5dc6cc1d91771f012/lua/flash/state.lua#L366C5-L366C41
where `opts.actions` should be just `actions`?

**Reply by folke** [reply]
Yes, you're right. Just pushed a fix.
Be aware that `actions` is still an internal API. Will probably change/document it at some point

**Reply by joalof** [reply]
Noted, thanks!
